plugin/vcs/gitlab/gitlab.go

```diff
@@ -281,12 +281,35 @@ func (p *Provider) ExchangeOAuthToken(ctx context.Context, instanceURL string, o
 	return oauthResp.toVCSOAuthToken(), nil
 }
 
-// FetchRepositoryList fetched all repositories in which the authenticated user
-// has a maintainer role.
-func (p *Provider) FetchRepositoryList(ctx context.Context, oauthCtx common.OauthContext, instanceURL string) ([]*vcs.Repository, error) {
+// FetchAllRepositoryList fetches all repositories in which the authenticated user has a maintainer role.
+func (p *Provider) FetchAllRepositoryList(ctx context.Context, oauthCtx common.OauthContext, instanceURL string) ([]*vcs.Repository, error) {
+	repoList := []*vcs.Repository{}
+	// page is the current requesting page index, starts from 1.
+	// perPage is the number of items to list per page (default: 20, max: 100).
+	// refer: https://docs.gitlab.com/ee/api/#offset-based-pagination
+	page, perPage := 1, 40
+
+	for {
+		temp, err := p.fetchOffsetRepositoryList(ctx, oauthCtx, instanceURL, page, perPage)
+		if err != nil {
+			return nil, err
```

Comment:
Maybe we should log an error and return the repo list we've already collected?

Reply:
* There is a logger from the method caller https://sourcegraph.com/github.com/bytebase/bytebase/-/blob/server/vcs.go?L177:74
* Maybe the incomplete list is not wanted: https://github.com/bytebase/bytebase/issues/1372

Reply:
SG.